### DISCUSSION TITLE
Is it possible to deploy OmniStream without using a domain?

### DISCUSSION THREAD

**Discussion by Xalcker** [kelinger/OmniStream#51] [Q&A] [answered]
I want to try OmniStream, but I don't see any option to not use a domain and access apps directly via ip:port on the local network. Can someone help me?

**Comment by TechPerplexed** [comment, marked as answer]
The thing is... OmniStream is MADE for use with a domain. The whole purpose of a domain is to make ports, internal IP addresses, protection, etc, dynamic. I guess you could theoretically use it without a domain, but you'd so severely cripple the product that it wouldn't be usable any longer. 

You can definitely use Omni on a local network though, many of us do. Splurge the $10/year or so for a domain, you won't regret it :)

**Reply by Xalcker** [reply]
Thank you for the reply, I understand your design, and it makes sense to use it that way, the problem is that at the moment I am locked in with T-Mobile's 5G internet and the modem/router they give does not allow opening ports to receive connections to devices on the local network.

**Reply by footlooseboss** [reply]
> device ... does not allow opening ports to receive connections to devices on the local network

Almost certainly 'carrier grade NAT' causing the issue (i.e. mobile/SIM service, not the modem device).

Tailscale should work, and depending on use case, [Tailscale funnel](https://tailscale.com/kb/1223/tailscale-funnel/)
